### PR TITLE
This removes libm for c and c++ and uses the clang headers for math f…

### PR DESCRIPTION
**Pull request by gregrodgers**
…unctions.  We still need a libm bitcode file to build FORTRAN device. So the code in __clang_hip_math.h can be used to build a libm device library for FORTRAN

This PR replaces PR160.

### CODE REVIEW

**Comment by gregrodgers** [comment]
This PR requires the pr in aomp-extras 
https://github.com/ROCm-Developer-Tools/aomp-extras/pull/31

File: clang/lib/Headers/__clang_hip_math.h [hunk 1]
```diff
@@ -665,521 +263,445 @@ inline double jn(int __n,
...
 // BEGIN INTRINSICS
-#if defined OCML_BASIC_ROUNDED_OPERATIONS
-__DEVICE__
-inline double __dadd_rd(double __x, double __y) {
-  return __ocml_add_rtn_f64(__x, __y);
+__DEVICE__ float __cosf(float x) { return __llvm_amdgcn_cos_f32(x); }
+__DEVICE__ float __exp10f(float x) { return __ocml_exp10_f32(x); }
+__DEVICE__ float __expf(float x) { return __ocml_exp_f32(x); }
+__DEVICE__ float __fadd_rd(float x, float y) {
+  return __ocml_add_rtp_f32(x, y);
```
Comment (aaronenyeshi) [inline clang/lib/Headers/__clang_hip_math.h:496]: Some of these rounding mode functions may cause you issues.
This one for example should be using `__ocml_add_rtn_f32` for `__fadd_rd`.

Here is my general understanding of these suffices matching CUDA to OCML.
- rd (round-down) = rtn (round-to-negative-infinity)
- rn (round-to-nearest-even) = rte (round-to-nearest-even)
- ru (round-up) = rtp (round-to-positive-infinity)
- rz (round-to-zero) = rtz (round-to-zero).

File: clang/lib/Headers/__clang_hip_math.h [hunk 2]
```diff
@@ -665,521 +263,445 @@ inline double jn(int __n,
...
-#pragma pop_macro("__DEF_FUN1")
-#pragma pop_macro("__DEF_FUN2")
-#pragma pop_macro("__DEF_FUNI")
-#pragma pop_macro("__DEF_FLOAT_FUN2I")
-#pragma pop_macro("__HIP_OVERLOAD1")
-#pragma pop_macro("__HIP_OVERLOAD2")
 #pragma pop_macro("__DEVICE__")
-#pragma pop_macro("__RETURN_TYPE")
+#pragma pop_macro("__DEVICE_VOID__")
+#pragma pop_macro("__FAST_OR_SLOW")
```
Comment (aaronenyeshi) [inline clang/lib/Headers/__clang_hip_math.h:705]: unused macro

**Review by aaronenyeshi — COMMENTED** [review]
Hi Greg, some of the hip header changes have landed upstream. I think you may want to rebase on top of latest hip headers. They have been updated to be very similar to cuda headers, and may make your life easier.

**Pull request merged**